WordPress/src/main/java/org/wordpress/android/ui/stats/datasets/StatsDatabaseHelper.java

```diff
@@ -0,0 +1,131 @@
+package org.wordpress.android.ui.stats.datasets;
+
+import android.content.Context;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+
+import org.wordpress.android.util.AppLog;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * database for all tracks information
+ */
+public class StatsDatabaseHelper extends SQLiteOpenHelper {
+    protected static final String DB_NAME = "stats.db";
+    private static final int DB_VERSION = 1;
+
+    /*
+	 *  database singleton
+	 */
+    private static StatsDatabaseHelper mDatabaseHelper;
+    private final static Object mDbLock = new Object();
+    private final Context mContext;
+
+    public static StatsDatabaseHelper getDatabase(Context ctx) {
+        if (mDatabaseHelper == null) {
+            synchronized(mDbLock) {
+                if (mDatabaseHelper == null) {
```

Comment:
This is done on purpose. It's the classic way of implementing singleton in multi-thread env.
The first if-condition (without the lock) checks if the DB wasn't initialized, in that case, the current thread gets the lock, checks again that the DB is still null (because another thread could have initialized it in the meanwhile) and initialize the DB.

Another way (slower) to do that same is to remove the first if-condition, and always checks within the lock. But the initialization phase is done once, while the call to `getDatabase` is executed a lot of time during the app live time.